File: src/DiameterOfBinaryTree.java
public class DiameterOfBinaryTree {
    int diameter = 0;

    public int diameterOfBinaryTree(TreeNode root) {
        computerDiameterOfBinaryTree(root);
        return diameter;
    }

    private int computerDiameterOfBinaryTree(TreeNode root) {
        if (root == null) return 0;
        int leftHeight = computerDiameterOfBinaryTree(root.left);
        int rightHeight = computerDiameterOfBinaryTree(root.right);
        diameter = Math.max(diameter, leftHeight + rightHeight);
        return 1 + Math.max(leftHeight, rightHeight);
    }
}

File: python/diameter_of_binary_tree.py
# Definition for a binary tree node.
class TreeNode:
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right


class Solution:
    def __init__(self):
        self.diameter = 0

    def computerDiameterOfBinaryTree(self, root: TreeNode) -> int:
        if root is None: return 0
        leftHeight = self.computerDiameterOfBinaryTree(root.left)
        rightHeight = self.computerDiameterOfBinaryTree(root.right)
        self.diameter = max(self.diameter, leftHeight + rightHeight)
        return 1 + max(leftHeight, rightHeight)

    def diameterOfBinaryTree(self, root: TreeNode) -> int:
        self.computerDiameterOfBinaryTree(root)
        return self.diameter
